src/app/technical-record/technical-record.component.ts

```diff
@@ -1,135 +1,119 @@
 
 @Component({
   selector: 'vtm-technical-record',
   templateUrl: './technical-record.component.html',
   styleUrls: ['../app.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush,
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class TechnicalRecordComponent implements OnInit, OnDestroy, ComponentCanDeactivate {
-
-  constructor(
-    private _store: Store<IAppState>,
-    public dialog: MatDialog,
-    private router: Router,
-    public techRecHelpers: TechRecordHelpersService,
-    public pendingChangesService: PendingChangesService) {
-
-    this.initializeTechnicalRecord();
-    this.navigationSubscription = this.router.events
-      .pipe(takeUntil(this.ngDestroyed$))
-      .subscribe((e: any) => {
-        if (e instanceof NavigationEnd) {
-          this.initializeTechnicalRecord();
-        }
-      });
-  }
-
-  @HostListener('window:beforeunload')
-  canDeactivate(): Observable<boolean> | boolean {
-    if (this.isFormDirty) {
-      return false;
-    } else { return true; }
-  }
-
+export class TechnicalRecordComponent implements OnInit {
   @HostBinding('@.disabled')
-  public animationsDisabled = true;
+  animationsDisabled = true;
 
-  ngDestroyed$ = new Subject();
-  techRecordsJson$: Observable<any>;
-  testResultJson$: Observable<any>;
-  isFormDirty: boolean;
+  editRecord: boolean;
+  showAdrDetails: boolean;
+  adrDisplayParams: { [key: string]: boolean };
+  techRecord: FormGroup;
 
   searchIdentifier = '{none searched}';
-  panels: { panel: string, isOpened: boolean }[] = [{ panel: 'panel1', isOpened: false }, { panel: 'panel2', isOpened: false },
-  { panel: 'panel3', isOpened: false }, { panel: 'panel4', isOpened: false },
-  { panel: 'panel5', isOpened: false }, { panel: 'panel6', isOpened: false }, { panel: 'panel7', isOpened: false },
-  { panel: 'panel8', isOpened: false }, { panel: 'panel9', isOpened: false }, { panel: 'panel10', isOpened: false }];
   allOpened = false;
-  color = 'red';
-  changeLabel = 'Change technical record';
-  isSubmit = false;
-  adrData = true;
-  hideForm = false;
-  showCheck = false;
-  numberFee: any;
-  isAdrNull: any;
-  navigationSubscription;
-
-  initializeTechnicalRecord() {
-    this.cancelAddrEdit();
-    this.techRecordsJson$ = this._store.select(selectVehicleTechRecordModelHavingStatusAll);
-    this.testResultJson$ = this._store.pipe(select(selectSelectedVehicleTestResultModel));
-  }
+  panels: { panel: string; isOpened: boolean }[] = [
+    { panel: 'panel1', isOpened: false },
+    { panel: 'panel2', isOpened: false },
+    { panel: 'panel3', isOpened: false },
+    { panel: 'panel4', isOpened: false },
+    { panel: 'panel5', isOpened: false },
+    { panel: 'panel6', isOpened: false },
+    { panel: 'panel7', isOpened: false },
+    { panel: 'panel8', isOpened: false },
+    { panel: 'panel9', isOpened: false },
+    { panel: 'panel10', isOpened: false }
+  ];
+
+  @Input() techRecordsJson: VehicleTechRecordModel[];
```

Comment:
we should no longer bring in VehicleTechRecordModel[] but rather VehicleTechRecordModel